platform-sdk/platform-apps/tests/AddressBookTestingTool/src/main/java/com/swirlds/demo/addressbook/AddressBookTestingToolState.java

```diff
@@ -243,17 +251,22 @@ public int getVersion() {
      */
     @Override
     @NonNull
-    public AddressBook updateWeight(@NonNull final AddressBook addressBook) {
+    public synchronized AddressBook updateWeight(
```

Comment:
Can this be called before init has been called? otherwise I do not see why we need to pass the context.

Reply:
This is exactly the problem... we must call this before we construct the `SwirldsPlatform` instance, but `init()` requires an instance of the `SwirldsPlatform` and therefore has to be called afterwards.

Reply:
The instance of `SwirldState` is not constructed until the platform is started.   `updateWeight()` needs to be called as part of the address book initialization before the platform is constructed.   In order to avoid adding static methods, this was the only way I saw to get the configuration settings to the `SwirldState `as a context for performing the `updateWeight()` operation.   I am open to hearing other solutions.